src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -409,16 +409,32 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
 
     fetchCallInputEntries(callKey.scope).map { entries =>
       entries.map { entry =>
+        // .get are used below because the exception will be captured by the Future
+        val taskInput = findTaskInput(entry.scope, entry.key.name).get
+
         val value = entry.wdlValue match {
-          case Some(e: WdlExpression) => e.evaluate(lookup, new NoFunctions).get
-          case Some(v) => v
-          case _ => throw new WdlExpressionException("Unknown error")
+          case Some(e: WdlExpression) => e.evaluate(lookup, new NoFunctions)
+          case Some(v) => Success(v)
+          case _ => Failure(new WdlExpressionException("Unknown error"))
         }
-        entry.key.name -> value
+        val coercedValue = value.map(x => taskInput.wdlType.coerceRawValue(x).get)
+        entry.key.name -> coercedValue.get
       }.toMap
     }
   }
 
+  private def findTaskInput(callFqn: String, outputName: String): Try[TaskInput] = {
```

Comment:
not sure but shouldn't it be `inputName` rather than `outputName` ? Also in the exception message.